source/Octopus.TestPortForwarder/SocketPump.cs

```diff
@@ -69,15 +91,34 @@ async Task PausePump(CancellationToken cancellationToken)
                 cancellationToken.ThrowIfCancellationRequested();
             }
         }
-
-        static async Task WriteToSocket(Socket writeTo, byte[] inputBuffer, int totalBytesToSend, CancellationToken cancellationToken)
+        
+        async Task WriteToSocketDelayingSendingTheLastNBytes(Socket writeTo, byte[] buffer, int bufferLength, int delaySendingLastNBytes, CancellationToken cancellationToken)
+        {
+            int howMuchToSend = bufferLength - delaySendingLastNBytes;
+            if(howMuchToSend < 0) howMuchToSend = bufferLength;
+            int sent = await WriteToSocket(writeTo, buffer, 0, howMuchToSend, cancellationToken);
+            if (howMuchToSend < bufferLength)
+            {
+                await Task.Delay(10);
```

Comment:
A possible future enhancement might be to configure this delay length. But I don't know if that's a requirement. Just an idea for later 😁

Reply:
It can be done when required.